### PR TITLE
scanner: Use Konf to parse the configuration file

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Use Konf [1] to parse the configuration file. Put the ScannerConfiguration
class in the model module because there will a unified configuration file
in future which needs to be parseable by all tools.

[1] https://github.com/uchuhimo/konf

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/707)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -78,10 +78,14 @@ subprojects {
         funTestCompile project(':utils-test')
     }
 
-    // TODO: Remove this once jackson-module-kotlin and kotlintest are updated.
     configurations.all {
         resolutionStrategy {
+            // TODO: Remove this once jackson-module-kotlin and kotlintest are updated.
             force 'org.jetbrains.kotlin:kotlin-reflect:1.2.51'
+
+            // Required because Konf dependens on version 1.21 which is incompatible with Jackson 1.9.x:
```
Comment (sschuberth) [inline build.gradle]: Isn't this supposed to say "Jackson 2.9.x"?

**Comment by mnonnenmacher** [comment]
@sschuberth Done.

File: build.gradle (hunk 1 above)
Comment (mnonnenmacher) [reply build.gradle]: Done.

File: scanner/src/test/kotlin/ScanResultsCacheTest.kt [hunk 2]
```diff
@@ -41,63 +41,18 @@ class ScanResultsCacheTest : WordSpec() {
 
     init {
         "ScanResultsCache.configure" should {
-            "fail if the cache type is missing" {
```
Comment (sschuberth) [inline scanner/src/test/kotlin/ScanResultsCacheTest.kt]: So, why exactly is it fine to drop all the tests for missing stuff now?

**Comment by mnonnenmacher** [comment]
Closing this PR as Konf does not meet all requirements. The main issue is that with the approach of using `objects` for the specs it is not trivial to serialize them.

**Pull request closed without merging**